adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -333,6 +336,21 @@ public AuthenticationResult getAuthTokenInBackground(final AuthenticationRequest
         return getResultFromBrokerResponse(bundleResult, request);
     }
 
+    private void setAndLogBrokerInstallationStatuses(final BrokerEvent brokerEvent) {
+        if (null == brokerEvent) {
+            Logger.d(TAG, "BrokerEvent was null. Proceeding.");
+            return;
+        }
+
+        final boolean isCompanyPortalInstalled = PackageHelper.isCompanyPortalInstalled(mContext.getPackageManager());
+        brokerEvent.setCompanyPortalInstalled(isCompanyPortalInstalled);
+        Logger.d(TAG, "Is Company Portal installed? [" + isCompanyPortalInstalled + "]");
```

Comment:
another note on how noisy this will be

Reply:
from my understanding, Samir asked for this specifically

---
In reply to: [168899043](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1122#discussion_r168899043) [](ancestors = 168899043)

Reply:
Confirming was requested by Teams (Rama)